File: src/main/java/com/bytebreakstudios/tictactoe/action/TieAction.java
package com.bytebreakstudios.tictactoe.action;

import com.bytebreakstudios.tictactoe.State;
import com.bytebreakstudios.board.Action;

public class TieAction extends Action<State> {

    @Override
    protected State innerApply(State state) {
        State next = (State) state.clone();
        next.winner = 0;
        next.finished = true;
        return next;
    }

    @Override
    protected State innerUnapply(State state) {
        State next = (State) state.clone();
        next.winner = 0;
        next.finished = false;
        return next;
    }

    @Override
    public String description() {
        return null;
    }

    @Override
    public String serialize() {
        return null;
    }

    @Override
    public Object deserialize(String s) {
        return null;
    }
}

File: src/main/scala/com/bytebreakstudios/tictactoe/scala/action/TieAction.scala
package com.bytebreakstudios.tictactoe.scala.action

import com.bytebreakstudios.board.Action
import com.bytebreakstudios.tictactoe.scala.State

class TieAction extends Action[State] {
  override def innerApply(t: State): State = t.copy(winner = 0, finished = true)

  override def innerUnapply(t: State): State = t.copy(winner = 0, finished = false)

  override def description(): String = ???

  override def serialize(): String = ???

  override def deserialize(s: String) = ???
}
